### DISCUSSION TITLE
deserializing a json array of different objects.

### DISCUSSION THREAD

**Discussion by wenzhengresideo** [stephenberry/glaze#298] [Q&A] [answered]
Hi how does glaze deal with the following problem?
[https://stackoverflow.com/questions/36479152/how-to-deserialize-a-json-array-containing-different-data-types-to-a-single-obje](https://stackoverflow.com/questions/36479152/how-to-deserialize-a-json-array-containing-different-data-types-to-a-single-obje)

[
  {
    "SEX": "Male",
    "BREED": "Opifex",
    "PVPRATING": 1301,
    "NAME": "Kilmanagh",
    "FIRSTNAME": "Big",
    "PVPTITLE": "Freshman",
    "LASTNAME": "Kahuna",
    "CHAR_DIMENSION": 5,
    "ALIENLEVEL": 30,
    "RANK_name": "Vindicator",
    "HEADID": 40281,
    "PROFNAME": "Guru",
    "LEVELX": 220,
    "PROF": "Martial Artist",
    "CHAR_INSTANCE": 12734,
    "SIDE": "Omni"
  },
  {
    "ORG_DIMENSION": 5,
    "RANK_TITLE": "President",
    "ORG_INSTANCE": 9911,
    "NAME": "Elements of Destruction",
    "RANK": 0
  },
  "2016/04/06 08:37:26"
]

**Comment by mwalcott3** [comment, marked as answer]
tldr use std::tuple or a meta with a glz::array value for this example case and std::vector<std::variant<ObjectTypes...>> generally. It should just work if all the objects have metas.

If the types always have the same order and length you can parse to a std::tuple of those types and it should work as long as they have metas for all the individual objects. ```std::tuple<Object1, Object2, std::string>```. Alternatively, you can also create an array based meta instead of an object one ```glz::array(&T::object1, &T::object2, &T::date_time)```

If the array has variable length and differing types you can create a std::vector of std::variants of the different object types https://github.com/stephenberry/glaze/blob/main/docs/variant-handling.md#auto-deduction-of-object-types and as long as the objects all have metas and they all have unique key combinations it should work automatically. ```std::vector<std::variant<Object1, Object2, std::string>>```

Worst case scenario you can parse to a ```std::vector<glz::json_t>``` but I would prefer using the variant support. glz::json_t exists for convivence and should be avoided when possible.

There are also some situations (rare) where you may want to utilize a two stage parse where you parse to an array of glz::raw_json_view and then do typed parses later when you are able to deduce the type.

If you need more details or a full example I can provide them.
